typing/ctype.ml

```diff
@@ -1937,7 +1937,7 @@ let rec unify_univar t1 t2 = function
 
 (* Test the occurrence of free univars in a type *)
 (* that's way too expensive. Must do some kind of caching *)
-let occur_univar env ty =
```

Comment:
This will sound familiar: I'm not keen on using an optional parameter here. All occurences of `occur_univar` are local to this file, we might as well add the argument to them all and avoid any risk of using the wrong version by accident.